internal/appsec/appsec.go

```diff
@@ -11,41 +11,56 @@ package appsec
 import (
 	"sync"
 
+	rc "github.com/DataDog/datadog-agent/pkg/remoteconfig/state"
+
 	"gopkg.in/DataDog/dd-trace-go.v1/internal/appsec/dyngo"
 	"gopkg.in/DataDog/dd-trace-go.v1/internal/log"
+	"gopkg.in/DataDog/dd-trace-go.v1/internal/remoteconfig"
 )
 
 // Enabled returns true when AppSec is up and running. Meaning that the appsec build tag is enabled, the env var
 // DD_APPSEC_ENABLED is set to true, and the tracer is started.
 func Enabled() bool {
 	mu.RLock()
 	defer mu.RUnlock()
-	return activeAppSec != nil
+	return activeAppSec != nil && activeAppSec.started
 }
 
 // Start AppSec when enabled is enabled by both using the appsec build tag and
 // setting the environment variable DD_APPSEC_ENABLED to true.
-func Start() {
-	enabled, err := isEnabled()
+func Start(opts ...StartOption) {
+	enabled, set, err := isEnabled()
 	if err != nil {
 		logUnexpectedStartError(err)
 		return
 	}
-	if !enabled {
+	// Check if AppSec is explicitly disabled
+	if set && !enabled {
 		log.Debug("appsec: disabled by the configuration: set the environment variable DD_APPSEC_ENABLED to true to enable it")
 		return
 	}
-
+	// From this point we know that AppSec is either enabled or can be enabled through remote config
 	cfg, err := newConfig()
 	if err != nil {
 		logUnexpectedStartError(err)
 		return
 	}
+	for _, opt := range opts {
+		opt(cfg)
+	}
 	appsec := newAppSec(cfg)
-	if err := appsec.start(); err != nil {
+
+	if !set {
+		// If the env var is not set AppSec is disabled but can be enabled through remote config
+		log.Debug("appsec: %s is not set. AppSec won't start until activated through remote configuration", enabledEnvVar)
+	} else if err := appsec.start(); err != nil { // AppSec is specifically enabled
 		logUnexpectedStartError(err)
 		return
 	}
+	if appsec.rc != nil {
+		// TODO: register ASM_FEATURES callback
+		go appsec.rc.Start()
```

Comment:
Sorry I just realized when seeing your 1-click enable PR that this is API is unusual in Go: `Start()` functions are usually responsible in doing whatever it takes to start the thing, meaning here that the goroutine should be hidden inside the `Start()` function.

So could you please simply remove that goroutine creation from here and put it in `Start()` around the ticker for loop?

PS: you see what you did with functions named `Run()` instead.